### DISCUSSION TITLE
how would I integrate an external watcher process?

### DISCUSSION THREAD

**Discussion by BrianHicks** [storybookjs/storybook#26957] [Help]
### Summary

I work on a big project where we build everything with Turborepo using fine-grained packages. So for example, our frontend package depends on the JS init, styles, images, etc, and each of those packages have their own dependencies. I want to set up Storybook to work within that. So I have two questions:

1. Is there a way to get Storybook to call `turbo build` in its package directory on changes? That will update all the static files it needs to look for in modules linked under `node_modules`.
2. Failing that, is there a way to get Storybook to run persistent processes so I can do it myself? I'd like this even aside from Storybook—for example, I'd like to configure [elm-watch](https://github.com/lydell/elm-watch) and still have a single entrypoint for folks to do `pnpm run storybook`.

This may be asking for a pony, but I don't see a lot about external process management in the docs so I figured I'd ask!

### Additional information

I'm starting from scratch, so using Storybook 8.0.9. I have a really minimal config, basically `storybook init` with the HTML/Vite preset.

```typescript
import { join, dirname } from "path"
import { StorybookConfig } from "@storybook/html-vite"

/**
 * This function is used to resolve the absolute path of a package.
 * It is needed in projects that use Yarn PnP or are set up within a monorepo.
 */
function getAbsolutePath(value) {
  return dirname(require.resolve(join(value, "package.json")))
}

const config: StorybookConfig = {
  stories: ["../stories/**/*.mdx", "../stories/**/*.stories.@(js|jsx|mjs|ts|tsx)"],
  addons: [
    getAbsolutePath("@storybook/addon-links"),
    getAbsolutePath("@storybook/addon-essentials"),
    getAbsolutePath("@chromatic-com/storybook"),
    getAbsolutePath("@storybook/addon-interactions"),
  ],
  framework: {
    name: getAbsolutePath("@storybook/html-vite") as "@storybook/html-vite",
    options: {},
  },
  docs: {
    autodocs: "tag",
  },
  typescript: {
    check: true,
    skipCompiler: false,
  },
}
export default config
```

**Comment by Marklb** [comment]
I think I have seen this sort of thing done, with Webpack, but I don't know where one of those examples is at the moment. I don't use Vite and I am not familiar enough with it to know how to do this, but if you know how to configure Vite then I would take a look at Storybook's `viteFinal` function. (https://storybook.js.org/docs/builders/vite#configuration) It should let you modify the config that Storybook is going to use. Depending on what you want to do, it may be difficult to modify or it may be as simple as dropping in a plugin.
